clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -313,6 +317,13 @@ private KafkaProducer(ProducerConfig config, Serializer<K> keySerializer, Serial
                 config.ignore(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG);
                 this.valueSerializer = valueSerializer;
             }
+
+            // load interceptors and make sure they get clientId
+            userProvidedConfigs.put(ProducerConfig.CLIENT_ID_CONFIG, clientId);
+            List<ProducerInterceptor<K, V>> interceptorList = (List) (new ProducerConfig(userProvidedConfigs)).getConfiguredInstances(ProducerConfig.INTERCEPTOR_CLASSES_CONFIG,
```

Comment:
Actually, since we are catching all interceptor exceptions, the behavior is when we call interceptor methods, we'll get exceptions which will be logged.